### ISSUE TITLE
[BUG] - waitForResponse hangs on forever if condition is false. Doesn't respect global timeout

### ISSUE THREAD

**Issue by AndreaPontrandolfo** [microsoft/playwright#17286]
**Context:**
- Playwright Version:@playwright/test": "^1.25.1 & playwright": "^1.25.1
- Operating System: Windows
- Node.js version: v16.10.0
- Browser: Chromium

**Code Snippet**

```
    const [response] = await Promise.all([
      page.waitForResponse(
        submittedFormResponse => {
          const requestMethod = submittedFormResponse.request().method();
          return requestMethod === 'POST';
        }
      ),
      submitForm(section),
    ]);
```

**Describe the bug**

waitForResponse hangs on forever if condition is false. Doesn't respect global timeout. It actually respect the timeout that you can optionally pass to waitForResponse, but if you omit it should fallback to the deafult command timeout, but it doesn't it just hangs on forever.

**Comment by mxschmitt** [comment]
waitForResponse will use the actionTimeout, see here: https://playwright.dev/docs/test-timeouts

Keep in mind that globalTimeout is different, globalTimeout is for your whole test execution to take.

To what is your actionTimeout configured?

**Comment by AndreaPontrandolfo** [comment]
this is what i have 
```
  timeout: 18 * 1000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 18 * 1000,
  },
```
and then 
```
  use: {
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
    actionTimeout: 0,
```

So i would guess that it hangs because my actionTimeout is set to 0, but why doesn't it fallback to the globalTimeout? Is it expected that it hangs indefinitely like this? Also considering that 0 is the default timeout for actions.
I feel that the whole timeouts configuring is a bit convoluted and sparse, in my opinion. 
I wouldn't expect Playwright to ever hang on like this.

**Comment by mxschmitt** [comment]
Because actionTimeout and globalTimeout are completely different things? See [here](https://playwright.dev/docs/api/class-testconfig#test-config-global-timeout) for globalTimeout, its for the whole test execution.

See also here again about all our timeouts, as an overview: https://playwright.dev/docs/test-timeouts

Keep in mind that if the test timeout is reached, that the test-runner will terminate the test and should print exactly where the waitForResponse was waiting as a codeframe.

If it does not, then its a bug!

Let's summarize, is it about that its using the wrong timeout, what I think, then its from our side as expected. Or is it that its about hanging and stalling somewhere for an infinite amount of time without good error reporting?

**Comment by AndreaPontrandolfo** [comment]
Yes thank you, i missed https://playwright.dev/docs/test-timeouts.
Explicitly setting actionTimeout solved the issue for me.
That being said, i'm not sure why some of these timeouts start with a default of 0/infinite. It doesn't look like the correct default, because then most people would end up in my same situation.
